File: VB/NorthwindDataProvider.vb
Imports System.Data
Imports System.Data.OleDb

Namespace RichEditImageMailMerge

    Public Module NorthwindDataProvider

        Private connectionString As String = "Provider=Microsoft.ACE.OLEDB.12.0;Data Source=|DataDirectory|\nwind.mdb;Persist Security Info=False;"

        Sub New()
            Using connection As OleDbConnection = New OleDbConnection(connectionString)
                Dim selectCommand As OleDbCommand = New OleDbCommand("SELECT * FROM Categories", connection)
                Dim da As OleDbDataAdapter = New OleDbDataAdapter(selectCommand)
                categoriesField = New DataTable("Categories")
                da.Fill(categoriesField)
                categoriesField.Constraints.Add("IDPK", categoriesField.Columns("CategoryID"), True)
                selectCommand.Dispose()
            End Using
        End Sub

        Private categoriesField As DataTable

        Public ReadOnly Property Categories As DataTable
            Get
                Return categoriesField
            End Get
        End Property
    End Module
End Namespace

File: CS/NorthwindDataProvider.cs
using System.Data;
using System.Data.OleDb;

namespace RichEditImageMailMerge {
    public static class NorthwindDataProvider {
        private static string connectionString = @"Provider=Microsoft.ACE.OLEDB.12.0;Data Source=|DataDirectory|\nwind.mdb;Persist Security Info=False;";

        static NorthwindDataProvider() {
            using (OleDbConnection connection = new OleDbConnection(connectionString)) {
                OleDbCommand selectCommand = new OleDbCommand("SELECT * FROM Categories", connection);
                OleDbDataAdapter da = new OleDbDataAdapter(selectCommand);

                categories = new DataTable("Categories");

                da.Fill(categories);

                categories.Constraints.Add("IDPK", categories.Columns["CategoryID"], true);
 
                selectCommand.Dispose();
            }
        }

        private static DataTable categories;

        public static DataTable Categories {
            get {
                return categories;
            }
        }
    }
}